plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -88,72 +140,129 @@ const Combobox = ( {
 		setValidationErrors,
 	] );
 
-	// @todo Remove patch for ComboboxControl once https://github.com/WordPress/gutenberg/pull/33928 is released
-	// Also see https://github.com/WordPress/gutenberg/pull/34090
+	const outerWrapperClasses = classnames(
+		'wc-block-components-combobox',
+		className || '',
+		{
+			'is-active': selectedOption?.value,
+			'has-error': error?.message && ! error?.hidden,
+		}
+	);
+
+	const ariaInvalid = error?.message && ! error?.hidden ? 'true' : 'false';
+	const store = useComboboxStore();
+	const activeValue = store.useState( 'activeValue' );
+
+	const onClose = useCallback( () => {
+		// If the search term doesn't match the selected option, try do an exact value match.
+		// e.g. if the user leaves "NZ" in the search box, select "New Zealand" on close.
+		// If not just leave the last selected value.
```

Comment:
I opted to leave one old behaviour out here, because its more complex and because I think it's worse behaviour but i am open to other opinions.

basically, if you open the combo, type some characters (e.g. "ss") and leave the combo it will select the closest match by label. I think it's quite heavy handed personally, and I opted instead to only do this for value matches as I have discussed in this comment here. If they type something like "ss", it will just go back to their last selection as it currently stands.

Reply:
I think the current behaviour is in line with what most users expect, actually. E.g. typing "Unit" then tabbing out doesn't usually select the closest match in other comboboxes i've used unless you arrow down to highlight the one you want.

I'm happy for the current functionality to remain as it is!

We can remove `findBestMatchByLabel` I think?

Reply:
@opr Thanks I agree. yes we can remove that function now :)

Reply:
Resolved in https://github.com/woocommerce/woocommerce/pull/46943/commits/93e79d34371bd65acd6e612c248b066d39596a94

Reply:
That's heavy handed indeed but it was the only way to get browser autofill to work.

Reply:
@senadir Yeah, I am seeing that perspective now that I dived deeper into the autocomplete. I have based on your comments gone deeper with it and tried to replicate old behaviour where it made sense, would love for you to have a look. Basically I have only applied heavy handed behaviour when the input is not in focus. I've also reintroduced autofill hints which greatly improved the situation as well.